ProcessMaker/screen-builder
# FOUR-5191 Refactor File download

## Issue & Reproduction Steps
Ticket: [FOUR-5191](https://processmaker.atlassian.net/browse/FOUR-5191)

Expected behavior: 
`_parent` should work for both the preview and file download components

Actual behavior: 
`_parent` does not work for both the preview and file download components

## Solution
- Remove the `pm4ConfigOverrides.requestFiles` variable that stored the uploaded file info and instead run an API Request to retrieve the file info for the file download component.
- Add support for `_parent` variable

## How to Test

<h3>Testing Notes:</h3>

The process QA attached to the ticket [FOUR-5191](https://processmaker.atlassian.net/browse/FOUR-5191) has incorrect usage of the file upload and download control. Based on [documentation](https://processmaker.gitbook.io/processmaker/designing-processes/design-forms/screens-builder/control-descriptions/file-download-control-settings) for the File Download control


>... participant can download a file to a local computer that was attached to the Request via a File Upload control in a **different Screen** in that Request.

Ensure that the file upload and file download controls are on different screens. Example: The process has the file upload in Task A and the file download in Task B. 

The File Preview control does not have the above requirement, the control should work on both the **same** AND **different** screen.

2. Ensure are on the correct branches listed in the **Related Tickets & Packages** section.
3. There are many different configurations/scenarios for the file download control, to help here is a [document](https://processmaker.slack.com/files/T02FG73JM/F034QTHMNQ0?origin_team=T02FG73JM) of some scenarios that you can test. 

Overall we need to test that the controls (File Download & File Preview) are working as originally intended and with the usage of the `_parent` variable. 

4. Test the file preview and download controls with the following packages: Files, WebEntry, Collections
5. Run the `FileDownload.spec.js` file.

## Related Tickets & Packages
- Core PR - https://github.com/ProcessMaker/processmaker/pull/4293
- Files Package PR - https://github.com/ProcessMaker/package-files/pull/70
- Web Entry Package PR - https://github.com/ProcessMaker/package-webentry/pull/136

## Code Review Checklist
